app/scripts/models/auth_brokers/fx-ios-v1.js

```diff
@@ -10,16 +10,39 @@
 define(function (require, exports, module) {
   'use strict';
 
-  var _ = require('underscore');
-  var FxDesktopV1AuthenticationBroker = require('models/auth_brokers/fx-desktop-v1');
+  const _ = require('underscore');
+  const p = require('lib/promise');
 
-  var proto = FxDesktopV1AuthenticationBroker.prototype;
+  const FxDesktopV1AuthenticationBroker = require('models/auth_brokers/fx-desktop-v1');
 
-  var FxiOSV1AuthenticationBroker = FxDesktopV1AuthenticationBroker.extend({
+  const proto = FxDesktopV1AuthenticationBroker.prototype;
+
+  const FxiOSV1AuthenticationBroker = FxDesktopV1AuthenticationBroker.extend({
     defaultCapabilities: _.extend({}, proto.defaultCapabilities, {
       chooseWhatToSyncCheckbox: false,
       convertExternalLinksToText: true
-    })
+    }),
+
+    _notifyRelierOfLogin: function (account) {
+      /**
+       * As a workaround for sign-in/sign-up confirmation view disappearing
+       * on iOS, delay the login message sent via the channel by 5 seconds.
+       * This will give the user an indication that they need to verify
+       * their email address.
+       */
+      const defer = p.defer();
+
+      this.window.setTimeout(() => {
```

Comment:
Instead of duplicating the logic from the ancestor's `_notifyRelierOfLogin` method, a simpler solution would be to just call the ancestor's method after a short delay:

```js
return p().delay(LOGIN_MESSAGE_DELAY_MS).then(() => {
  return proto._notifyRelierOfLogin.call(this, account);
});
```